common/main/java/com/couchbase/lite/Collection.java

```diff
@@ -0,0 +1,189 @@
+package com.couchbase.lite;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.Executor;
+
+import com.couchbase.lite.internal.utils.Preconditions;
+
+
+public class Collection implements Indexable, DatabaseChangeObservable {
+    private final String name;
+    private final int count;
+    private final Scope scope;
+    private final Database database;
+
+    public Collection(@NonNull String name, int count, @NonNull Scope scope, @NonNull Database database) {
+        this.name = name;
+        this.count = count;
+        this.scope = scope;
+        this.database = database;
+    }
+
+    /**
+     * Gets an existing Document object with the given ID. If the document with the given ID doesn't
+     * exist in the collection, the value returned will be null.
+     */
+    @Nullable
+    public Document getDocument(@NonNull String id) { return database.getDocument(id); }
+
+    /**
+     * Save a document into the collection. The default concurrency control, lastWriteWins,
+     * will be used when there is conflict during  save.
+     * <p>
+     * When saving a document that already belongs to a collection, the collection instance of
+     * the document and this collection instance must be the same, otherwise, the InvalidParameter
+     * error will be thrown.
+     */
+    public void save(@NonNull MutableDocument document) throws CouchbaseLiteException {
+        database.save(document, ConcurrencyControl.LAST_WRITE_WINS);
+    }
+
+    /**
+     * Save a document into the collection with a specified concurrency control. When specifying
+     * the failOnConflict concurrency control, and conflict occurred, the save operation will fail with
+     * 'false' value returned.
+     * When saving a document that already belongs to a collection, the collection instance of the
+     * document and this collection instance must be the same, otherwise, the InvalidParameter
+     * error will be thrown.
+     */
+    public boolean save(@NonNull MutableDocument document, @NonNull ConcurrencyControl concurrencyControl)
```

Comment:
Cute, but I think that if the signature takes multiple lines, you have to put the body on a separate line.